core/src/mender-client.c

```diff
@@ -1139,7 +1139,7 @@ mender_client_update_work_function(void) {
     {
         cJSON *deployment_id_j = cJSON_GetObjectItem(mender_client_deployment_data, "id");
         if (NULL != deployment_id_j) {
-            deployment_id = deployment_id_j->valuestring;
+            deployment_id = cJSON_GetStringValue(deployment_id_j);
```

Comment:
This is questionable, I think. The internal data structure is not really internal, it's part of the header and thus a part of the public API. A function call comes with an overhead. Accessing the field directly skips any type checking, but it's guaranteed to be a JSON string, right?